// auto-reload-brunch/lib/index.js
// Generated by CoffeeScript 1.3.3
(function() {
  var AutoReloader, WebSocketServer, sysPath,
    __bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; };

  sysPath = require('path');

  WebSocketServer = (require('ws')).Server;

  module.exports = AutoReloader = (function() {

    AutoReloader.prototype.brunchPlugin = true;

    function AutoReloader(config) {
      var _this = this;
      this.config = config;
      this.onCompile = __bind(this.onCompile, this);

      this.connections = [];
      if (this.config.persistent) {
        this.server = new WebSocketServer({
          host: '0.0.0.0',
          port: 9485
        });
        this.server.on('connection', function(connection) {
          _this.connections.push(connection);
          return connection.on('close', function() {
            return _this.connections.splice(connection, 1);
          });
        });
      }
    }

    AutoReloader.prototype.onCompile = function(changedFiles) {
      var allCss, message,
        _this = this;
      if (!this.config.persistent) {
        return;
      }
      allCss = (changedFiles.length > 0) && (changedFiles.every(function(file) {
        return file.type === 'stylesheet';
      }));
      message = allCss ? 'stylesheet' : 'page';
      return this.connections.filter(function(connection) {
        return connection.readyState === 1;
      }).forEach(function(connection) {
        return connection.send(message);
      });
    };

    AutoReloader.prototype.include = [sysPath.join(__dirname, '..', 'vendor', 'auto-reload.js')];

    return AutoReloader;

  })();

}).call(this);
